src/material/tabs/paginated-tab-header.ts

```diff
@@ -63,6 +63,12 @@ export type ScrollDirection = 'after' | 'before';
  */
 const EXAGGERATED_OVERSCROLL = 60;
 
+/**
+ * The distance in pixels added on top of the scroll amount when using paginations to scroll a
+ * tab list. It is used to compensate for the padding on either side of a label.
+ */
+const REDEEMING_OVERSCROLL = 24;
+
```

Comment:
Playing around with the new code, it seems like now users have to click more times to reveal the next tab, because the first time only scrolls a few pixels. You can debug this in a browser by running `yarn dev-app`. See:

https://user-images.githubusercontent.com/4450522/223483953-31be97aa-87d6-44f7-9397-2b9c4fc3407e.mov

Reply:
Yes, the reason for that is it guarded the scroll amount by the space remaining on the left side when you did that. I was thinking that as an expected behavior. 

We could do that logic only when some labels are clipped either on the left or right, but that requires a more specific implementation instead of like the current one which works for all the labels. What we might need to do is first finding the label(s) in that circumstance, and use the logic to guard its move, but leave it the same for all other labels.

Reply:
Hmm, not sure that I follow how the proposed approach would work. This might be one of those cases where we play around with a couple of approaches to see which one feels the most intuitive. 

A couple approaches off the top of my head:
* Find the first tab that is entirely out of the view and scroll the viewport so the entire label is visible. This one would work well for a narrow viewport, but it might be annoying on a larger one.
* Figure out how many tabs fit in a "page" worth of pagination and scroll the entire next page into view. E.g. in my example it would scroll 3 tabs at a time. I could imagine this being a little jarring on a very wide tab group though.

Reply:
> Find the first tab that is entirely out of the view and scroll the viewport so the entire label is visible. This one would work well for a narrow viewport, but it might be annoying on a larger one.

There two problems with this approach,
1. We are not sure if a label can be displayed entirely, what we can do is to scroll the label so that either the left or right border of the label is on the left or right border of the viewport, to make sure it could be fully visible if the length allows.
2. It does not make the clipped label fully visible. Let's say a label is pretty long and always get cut either left or right (please see example below), every time we scroll we would pick the first **entirely invisible** label from either side and scroll that to view. It would cause the partially clipped label ("Placeholder for very long label" in the example) to be moved out of the view.
![MatTab-reflow](https://user-images.githubusercontent.com/20229878/223562347-85cffe39-97fb-4308-ad49-49353431dbdc.gif)

> Figure out how many tabs fit in a "page" worth of pagination and scroll the entire next page into view. E.g. in my example it would scroll 3 tabs at a time. I could imagine this being a little jarring on a very wide tab group though.

This probably would work, but how to decide what fits in a page is a problem. Do we consider "page" as the labels that could fully displayed? Then it will work the same as my original approach in this commit and cause the same problem.

While I am responding I start to realize the problem you brought up above is actually caused by the inconsistence between the two strategies we applied to move the tab label (1/3 of the viewport, and the one I introduced), if we use any of them exclusively it would not be an issue. Say, when we click 'before', we always move the next label to make its right border aligned with the right border of the viewport; when we click 'after', we always move the next label to make its left border aligned with the left border of the viewport.  This way we would not have those scrolls with only a few pixels in most cases.